### PR TITLE
[dif] simplify autogen DIF templates by sharing typedefs

### PR DESCRIPTION
**Pull request by timothytrippel**
This partially addressed #8142 by sharing, rather than autogenerating,
duplicate common DIFs typedefs.

### CODE REVIEW

File: sw/device/lib/dif/dif_base.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * ground.
+ *
+ * Normally, the standard way to drop such a value on the ground explicitly is
+ * with the syntax `(void)expr;`, in analogy with the behavior of C++'s
+ * `[[nodiscard]]` attribute.
+ * However, GCC does not implement this, so the idiom `if (expr) {}` should be
+ * used instead, for the time being.
+ * See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=25509.
+ */
+#define DIF_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
```
Comment (mcy) [inline sw/device/lib/dif/dif_base.h:31]: I'd just move this into `macros.h`. A lot of other places want to use this and thus re-define this macro.

File: sw/device/lib/dif/dif_warn_unused_result.h [hunk 2]
```diff
@@ -5,6 +5,9 @@
 #ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_WARN_UNUSED_RESULT_H_
 #define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_WARN_UNUSED_RESULT_H_
 
+// WARNING: This file will soon be deprecated, since this definition is included
```
Comment (mcy) [inline sw/device/lib/dif/dif_warn_unused_result.h:8]: Just delete the contents of the file and #include the new one.

If we move the macro to macros.h though we can leave this as is (or define DIF_... in terms of the new macro).

File: sw/device/lib/dif/dif_base.h [hunk 3]
```diff
@@ -0,0 +1,83 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_BASE_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_BASE_H_
+
+/**
+ * @file
+ * @brief Unused Result Warning Macro for DIFs.
```
Comment (mcy) [inline sw/device/lib/dif/dif_base.h:10]: I think you missed a comment. ;)

File: sw/device/lib/dif/dif_base.h [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ * with the syntax `(void)expr;`, in analogy with the behavior of C++'s
+ * `[[nodiscard]]` attribute.
+ * However, GCC does not implement this, so the idiom `if (expr) {}` should be
+ * used instead, for the time being.
+ * See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=25509.
+ */
+#define DIF_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
+
+/**
+ * The result of a ${ip.name_long_lower} operation.
```
Comment (mcy) [inline sw/device/lib/dif/dif_base.h:34]: This one too, hehe.

File: sw/device/lib/dif/dif_base.h [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * NOTE: additional result values can be defined in the manually-implemented
```
Comment (mcy) [inline sw/device/lib/dif/dif_base.h:36]: How is this supposed to work? I think your doc briefly touched on this, but maybe doc'ing it here is valuable.

**Review by mcy — COMMENTED** [review]
LG, but I want to sort out the long-term home of the must_use macro.

**Review by mcy — APPROVED** [review]
*Plunk*

**Review by sriyerg — APPROVED** [review]

**Pull request merged**